### PR TITLE
Remove Koin and use Hilt instead.

### PR DESCRIPTION
**Pull request by caiofaustino**
This change only affects the Example App.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -28,6 +29,8 @@ buildscript {
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath "gradle.plugin.com.github.spotbugs.snom:spotbugs-gradle-plugin:$spotbugs_gradle_plugin_version"
         classpath "org.jetbrains.dokka:dokka-gradle-plugin:$dokka_version"
+        classpath 'com.google.dagger:hilt-android-gradle-plugin:2.38.1'
```
Comment (ozgur00) [inline build.gradle:32]: this line looks unnecessary

Comment (caiofaustino) [reply build.gradle:32]: Ahh thx, I didn't see it was duplicated.

**Review by ozgur00 — APPROVED** [review]

File: example-app/src/main/java/com/adyen/checkout/example/di/NetworkModule.kt [hunk 2]
```diff
@@ -15,27 +15,42 @@ import com.adyen.checkout.example.data.api.CheckoutApiService
...
+object NetworkModule {
 
-    fun provideHttpClient(): OkHttpClient {
-        val builder = OkHttpClient.Builder().let {
+    private val BASE_URL = if (CheckoutApiService.isRealUrlAvailable())
+        BuildConfig.MERCHANT_SERVER_URL
+    else
+        "http://myserver.com/my/endpoint/"
+
+    @Singleton
```
Comment (jreij) [inline example-app/src/main/java/com/adyen/checkout/example/di/NetworkModule.kt:42]: Do all these methods have to be singletons?

Comment (caiofaustino) [reply example-app/src/main/java/com/adyen/checkout/example/di/NetworkModule.kt:42]: Not really, I think it's good practice for the Retrofit instance to be a singleton that's reused by the repositories, and the repos are also singletons usually, so idk if there would be any benefit of trying to have them cleared from memory at some point.

**Pull request merged**